pkg/policy/mapstate.go

```diff
@@ -892,22 +877,13 @@ func (ms *mapState) RemoveDependent(owner Key, dependent Key, identities Identit
 
 // Merge adds owners, dependents, and DerivedFromRules from a new 'entry' to an existing
 // entry 'e'. 'entry' is not modified.
-// IsDeny, ProxyPort, and AuthType are merged by giving precedence to deny over non-deny, proxy
-// redirection over no proxy redirection, and explicit auth type over default auth type.
+// Merge is only called if both entries are allow or deny entries, so deny precedence is not
```

Comment:
Can we enforce this with an assertion / panic?

Reply:
No asserts or panics in production code, so the best we can do here IMO is to bail out and log a warning.